vm/src/vm.rs

```diff
@@ -160,6 +179,7 @@ impl VirtualMachine {
             trace_func,
             use_tracing: RefCell::new(false),
             settings,
+            signal_handlers: RefCell::new(HashMap::new()),
```

Comment:
You could use `Default::default()` here instead of `RefCell::new(HashMap::new())`